common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java

```diff
@@ -4,14 +4,16 @@
 import net.minecraft.resources.ResourceLocation;
 
 public final class ResourceLocations {
+    private static final ResourceLocation BINGO_PROTOTYPE = ResourceLocation.fromNamespaceAndPath(Bingo.MOD_ID, "");
+
     private ResourceLocations() {
     }
 
     public static ResourceLocation bingo(String path) {
-        return new ResourceLocation(Bingo.MOD_ID, path);
+        return BINGO_PROTOTYPE.withPath(path);
```

Comment:
Hmm, why not just use `ResourceLocation.withNamespaceAndPath` directly?

Reply:
Or just partially apply the function `ResourceLocation.fromNamespaceAndPath` if you really want that prototype.

Reply:
I'm doing this to cache the validity of the namespace. Every time you run `fromNamespaceAndPath`, it rechecks if the namespace is valid. `fromPath` avoids this check.

Reply:
Should probably add a comment then because that's not clear